File: main/cut-the-sticks/cut-the-sticks2.py
#!/usr/bin/env python3

input() # don't need n
heights = sorted(map(int, input().split()))
countdown = range(len(heights), 0, -1)

floor = 0
for height, remaining in zip(heights, countdown):
    if floor < height:
        print(remaining)
        floor = height

File: main/cut-the-sticks/cut-the-sticks2.pl
#!/usr/bin/env perl

use strict;
use warnings;

<>; # don't need n
my @heights = sort { $a <=> $b } split /\s+/msx, <>;

my $floor = 0;
for my $i (0..$#heights) {
    if ($floor < $heights[$i]) {
        print scalar(@heights) - $i, "\n";
        $floor = $heights[$i];
    }
}
